src/StatusBar.ts

```diff
@@ -92,20 +95,20 @@ export class StatusBar {
   }
   bind(source: string) {
     return {
-      initial: () => {
-        this.request(source, 'initial')
+      initial: (modes?: Mode[]) => {
+        this.request(source, 'initial', undefined, modes)
       },
-      running: (details?: string) => {
-        this.request(source, 'running', details)
+      running: (details?: string, modes?: Mode[]) => {
+        this.request(source, 'running', details, modes)
       },
-      success: (details?: string) => {
-        this.request(source, 'success', details)
+      success: (details?: string, modes?: Mode[]) => {
+        this.request(source, 'success', details, modes)
       },
-      failed: (details?: string) => {
-        this.request(source, 'failed', details)
+      failed: (details?: string, modes?: Mode[]) => {
+        this.request(source, 'failed', details, modes)
       },
-      stopped: (details?: string) => {
-        this.request(source, 'stopped', details)
+      stopped: (details?: string, modes?: Mode[]) => {
+        this.request(source, 'stopped', details, modes)

```

Comment:
This pattern feels a bit strange. What about instead adding a method to JestExt which calls `getStatusModes` and `request`?

Reply:
> What about instead adding a method to JestExt which calls getStatusModes and request?
right now mode and status are combined into a single string, if we make them separate calls, extra logic is needed to remember the missing info.

However, having said that, we could simplify the API by removing modes from all binding method except `initial` and injecting them in the wrapper within...

Reply:
turns out the new API I was thinking about is even more cumbersome... maybe I should understand a bit more your concern so we can come up with a better approach...

Reply:
My approach would have been https://github.com/connectdotz/vscode-jest/pull/1.